packages/lib/src/vis/shared/SelectionLine.tsx

```diff
@@ -31,23 +17,32 @@ function SelectionLine(props: Props) {
     color = 'black',
   } = props;
 
-  const { dataToWorld } = useAxisSystemContext();
-  const startPoint = dataToWorld(dataStartPoint);
-  const endPoint = dataToWorld(dataEndPoint);
+  const { width, height } = useThree((state) => state.size);
 
-  const [dataGeometry] = useState(() => new BufferGeometry());
-  const invalidate = useThree((state) => state.invalidate);
+  const { dataToWorld } = useAxisSystemContext();
+  const worldToHtml = useWorldToHtml();
 
-  useLayoutEffect(() => {
-    dataGeometry.setFromPoints([startPoint, endPoint]);
-    dataGeometry.computeBoundingSphere();
-    invalidate();
-  }, [dataGeometry, endPoint, invalidate, startPoint]);
+  const worldStartPoint = dataToWorld(dataStartPoint);
+  const htmlStartPt = worldToHtml(
+    new Vector3(worldStartPoint.x, worldStartPoint.y, 0)
+  );
+  const worldEndPoint = dataToWorld(dataEndPoint);
+  const htmlEndPt = worldToHtml(
+    new Vector3(worldEndPoint.x, worldEndPoint.y, 0)
+  );
```

Comment:
I think we could improve this. For example, making `dataToWorld` return a `Vector3` when given a `Vector3` would remove the need of the conversion to `Vector3` in the selection component.

Your thoughts ?

Reply:
How about changing `worldToHtml` to accept a `Vector2` instead of a `Vector3` and adapt the code in `Html` accordingly?

Reply:
Perhaps this is a bit tricky due to `Vector2` not having a `project` method, but I'm thinking that you could make a new `Vector3` in `worldToHtml` with `0` as `z` and convert back to `Vector2` before returning.